spinedb_api/db_mapping_base.py

```diff
@@ -500,10 +547,16 @@ def remove_unique(self, item):
     def _make_and_add_item(self, item):
         if not isinstance(item, MappedItemBase):
             item = self._make_item(item)
+            item.convert_db_ids_to_item_ids()
```

Comment:
Same as above, I find it a bit funny that we need to replace all references to db ids with our temp ids on all the items. Before we would keep the ids intact and find the reference by looking at the id map. Maybe we could consider going back to that? It would require less code and less processing.